tests/PHPUnit/System/ArchiveInvalidationTest.php

```diff
@@ -201,7 +203,103 @@ public function testDisablePluginArchiveBySiteId()
         $this->assertFalse(Rules::isSegmentPluginArchivingDisabled('testPlugin', 1));
     }
 
+    /**
+     * Check that dates are correctly parsed without any exceptions
+     */
+    public function getDatesToParse(): iterable
+    {
+        yield 'normal range' => [
+            '2020-01-01,2020-03-31',
+            'range',
+            [['2020-01-01,2020-03-31'], []],
+        ];
+
+        yield 'range using lastX keyword' => [
+            'last30',
+            'range',
+            [[Date::factory('now')->subDay(29)->toString() . ',' . Date::factory('now')->toString()], []],
+        ];
+
+        yield 'single day' => [
+            '2020-01-01',
+            'day',
+            [['2020-01-01'], []],
+        ];
+
+        yield 'single month' => [
+            '2020-04-01',
+            'month',
+            [['2020-04-01'], []],
+        ];
+
+        yield 'multiple days' => [
+            '2020-05-01,2020-05-02',
+            'day',
+            [[Date::factory('2020-05-01'), Date::factory('2020-05-02')], []],
+        ];
+
+        yield 'magic day keywords' => [
+            'today,yesterday',
+            'period' => 'day',
+            [[Date::factory('today'), Date::factory('yesterday')], []],
+        ];
+
+        yield 'invalid text' => [
+            'abcdef',
+            'period' => 'day',
+            [[], ['abcdef']],
+        ];
+
+        yield 'invalid number' => [
+            167389494,
+            'period' => 'day',
+            [[], [167389494]],
+        ];
+
+        yield 'invalid day' => [
+            '2020-02-31',
+            'period' => 'day',
+            [[], ['2020-02-31']],
+        ];
+
+        yield 'invalid range' => [
+            '2020-08-15, 2020-08-01',
+            'period' => 'range',
+            [[], ['2020-02-31']],
+            true
+        ];
+    }
+
+    /**
+     * @dataProvider getDatesToParse
+     */
+    public function testDatesCorrectlyParsed($dateString, $period, $expected, bool $expectAPIError = false)
+    {
+        // Test API
+        $r = new Request("module=API&method=CoreAdminHome.invalidateArchivedReports&idSites=".self::$fixture->idSite1."&period=" .
+            $period . "&dates=" . $dateString);
+        if ($expectAPIError) {
+            $this->assertApiResponseHasError($r->process());
+        } else {
+            $this->assertApiResponseHasNoError($r->process());
+        }
 
+        // Test date parsing method
+        $api = CoreAdminHomeApi::getInstance();
+        $reflection = new \ReflectionClass(CoreAdminHomeApi::class);
+        $method = $reflection->getMethod('getDatesToInvalidateFromString');
+        $method->setAccessible(true);
+
+        $parameters = [$dateString, $period];
+        try {
+            $result = $method->invokeArgs($api, $parameters);
+            self::assertEquals($expected, $result);
+        } catch (\Exception $e) {
+            if (!$expectAPIError) {
+                throw $e;
+            }
+        }
```

Comment:
There is a PHPUnit feature you could maybe use to handle that with less complexity:
```suggestion
        if ($expectAPIError) {
            self::expectException(\Exception::class);
        }

        $result = $method->invokeArgs($api, $parameters);
        self::assertEquals($expected, $result);
```